### ISSUE TITLE
NPE in ComputedCSSStyleDeclaration.getLeft

### ISSUE THREAD

**Issue by gaellalire** [HtmlUnit/htmlunit#488]
```java
Caused by: java.lang.NullPointerException: null
	at com.gargoylesoftware.htmlunit.javascript.host.css.ComputedCSSStyleDeclaration.getLeft(ComputedCSSStyleDeclaration.java:1580)
	at com.gargoylesoftware.htmlunit.javascript.host.html.HTMLElement.getOffsetLeft(HTMLElement.java:1393)
```
Issue is still there in 2.62.0.

I tested if parent is null to workaround the issue, worked for me but maybe not the best fix.

```java
			final HTMLElement parent = (HTMLElement) getElement().getParentElement();
			if (parent != null) {
				final ComputedCSSStyleDeclaration style = parent.getWindow().getComputedStyle(parent, null);
				left = pixelValue(style.getLeftWithInheritance());
			} else {
				left = 0;
			}
```

**Comment by rbri** [comment]
> Issue is still there in 2.62.0.

Was there another issue before?

**Comment by rbri** [comment]
Your code is calling offsetLeft on an element without a parent - correct?

**Comment by gaellalire** [comment]
> Was there another issue before?

2.60.0 is also impacted

> Your code is calling offsetLeft on an element without a parent - correct?

No idea, I'm using htmlunit-driver for selenium request. I did not check what the website is producing. Here is the full stacktrace
```java 
com.gargoylesoftware.htmlunit.ScriptException: Exception invoking getClientX
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine$HtmlUnitContextAction.run(JavaScriptEngine.java:980)
	at net.sourceforge.htmlunit.corejs.javascript.Context.call(Context.java:582)
	at net.sourceforge.htmlunit.corejs.javascript.ContextFactory.call(ContextFactory.java:481)
	at com.gargoylesoftware.htmlunit.javascript.HtmlUnitContextFactory.callSecured(HtmlUnitContextFactory.java:349)
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine.callFunction(JavaScriptEngine.java:907)
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine.callFunction(JavaScriptEngine.java:879)
	at com.gargoylesoftware.htmlunit.html.HtmlPage.executeJavaScriptFunction(HtmlPage.java:2613)
	at com.gargoylesoftware.htmlunit.html.HtmlPage.executeJavaScriptFunction(HtmlPage.java:2606)
	at com.gargoylesoftware.htmlunit.javascript.host.event.EventListenersContainer.executeEventListeners(EventListenersContainer.java:342)
	at com.gargoylesoftware.htmlunit.javascript.host.event.EventListenersContainer.executeAtTargetListeners(EventListenersContainer.java:379)
	at com.gargoylesoftware.htmlunit.javascript.host.event.EventTarget.fireEvent(EventTarget.java:173)
	at com.gargoylesoftware.htmlunit.html.DomElement.lambda$fireEvent$0(DomElement.java:1490)
	at net.sourceforge.htmlunit.corejs.javascript.Context.call(Context.java:582)
	at net.sourceforge.htmlunit.corejs.javascript.ContextFactory.call(ContextFactory.java:481)
	at com.gargoylesoftware.htmlunit.javascript.HtmlUnitContextFactory.callSecured(HtmlUnitContextFactory.java:349)
	at com.gargoylesoftware.htmlunit.html.DomElement.fireEvent(DomElement.java:1490)
	at com.gargoylesoftware.htmlunit.html.DomElement.doMouseEvent(DomElement.java:1433)
	at com.gargoylesoftware.htmlunit.html.DomElement.mouseDown(DomElement.java:1336)
	at com.gargoylesoftware.htmlunit.html.DomElement.click(DomElement.java:976)
	at org.openqa.selenium.htmlunit.HtmlUnitWebElement.submitForm(HtmlUnitWebElement.java:194)
	at org.openqa.selenium.htmlunit.HtmlUnitWebElement.submitImpl(HtmlUnitWebElement.java:151)
	at org.openqa.selenium.htmlunit.HtmlUnitKeyboard.sendKeys(HtmlUnitKeyboard.java:62)
	at org.openqa.selenium.htmlunit.HtmlUnitDriver.lambda$sendKeys$6(HtmlUnitDriver.java:341)
	at org.openqa.selenium.htmlunit.HtmlUnitDriver.lambda$runAsync$0(HtmlUnitDriver.java:292)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.RuntimeException: Exception invoking getClientX
	at net.sourceforge.htmlunit.corejs.javascript.MemberBox.invoke(MemberBox.java:238)
	at net.sourceforge.htmlunit.corejs.javascript.AccessorSlot$MemberBoxGetter.getValue(AccessorSlot.java:107)
	at net.sourceforge.htmlunit.corejs.javascript.AccessorSlot.getValue(AccessorSlot.java:67)
	at net.sourceforge.htmlunit.corejs.javascript.ScriptableObject.get(ScriptableObject.java:244)
	at com.gargoylesoftware.htmlunit.javascript.HtmlUnitScriptable.get(HtmlUnitScriptable.java:124)
	at net.sourceforge.htmlunit.corejs.javascript.ScriptableObject.getProperty(ScriptableObject.java:2031)
	at net.sourceforge.htmlunit.corejs.javascript.ScriptRuntime.getObjectElem(ScriptRuntime.java:1641)
	at net.sourceforge.htmlunit.corejs.javascript.ScriptRuntime.getObjectElem(ScriptRuntime.java:1624)
	at net.sourceforge.htmlunit.corejs.javascript.Interpreter.doGetElem(Interpreter.java:2724)
	at net.sourceforge.htmlunit.corejs.javascript.Interpreter.interpretLoop(Interpreter.java:1591)
	at net.sourceforge.htmlunit.corejs.javascript.Interpreter.interpret(Interpreter.java:1053)
	at net.sourceforge.htmlunit.corejs.javascript.InterpretedFunction.call(InterpretedFunction.java:92)
	at net.sourceforge.htmlunit.corejs.javascript.ContextFactory.doTopCall(ContextFactory.java:389)
	at com.gargoylesoftware.htmlunit.javascript.HtmlUnitContextFactory.doTopCall(HtmlUnitContextFactory.java:335)
	at net.sourceforge.htmlunit.corejs.javascript.ScriptRuntime.doTopCall(ScriptRuntime.java:3913)
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine$3.doRun(JavaScriptEngine.java:900)
	at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine$HtmlUnitContextAction.run(JavaScriptEngine.java:963)
	... 26 common frames omitted
Caused by: java.lang.NullPointerException: null
	at com.gargoylesoftware.htmlunit.javascript.host.css.ComputedCSSStyleDeclaration.getLeft(ComputedCSSStyleDeclaration.java:1580)
	at com.gargoylesoftware.htmlunit.javascript.host.html.HTMLElement.getOffsetLeft(HTMLElement.java:1393)
	at com.gargoylesoftware.htmlunit.javascript.host.html.HTMLElement.getPosX(HTMLElement.java:1431)
	at com.gargoylesoftware.htmlunit.javascript.host.event.MouseEvent.getScreenX(MouseEvent.java:243)
	at com.gargoylesoftware.htmlunit.javascript.host.event.MouseEvent.getClientX(MouseEvent.java:219)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at net.sourceforge.htmlunit.corejs.javascript.MemberBox.invoke(MemberBox.java:215)
	... 42 common frames omitted
```

**Comment by rbri** [comment]
ah ok, will have a look

**Comment by rbri** [comment]
Looks so simple but i'm struggling to build a test case for this.
Have to think about this.

Can you please try with the brand new 2.63.0 release i did yesterday and check if this is still there.

**Comment by gaellalire** [comment]
Still same with 2.63.0. I made a Maven project for you to reproduce.

**Comment by rbri** [comment]
OK, many thanks, running your code inside the debugger provides the final hint. Now i can reproduce your case.
Will try to fix it and inform via twitter if a new snapshot is available.

**Comment by rbri** [comment]
fixed, please try 2.64.0-SNAPSHOT.
